pkg/image/graph/nodes/nodes.go

```diff
@@ -161,12 +161,27 @@ func FindOrCreateSyntheticImageStreamNode(g osgraph.MutableUniqueGraph, is *imag
 	).(*ImageStreamNode)
 }
 
-// EnsureImageLayerNode adds a graph node for the layer if it does not already exist.
-func EnsureImageLayerNode(g osgraph.MutableUniqueGraph, layer string) graph.Node {
-	return osgraph.EnsureUnique(g,
-		ImageLayerNodeName(layer),
+// EnsureImageComponentNode adds a graph node for the image config or layer if it does not already exist.
+func EnsureImageComponentNode(g osgraph.MutableUniqueGraph, name string, t ImageComponentType) graph.Node {
```

Comment:
> The reasoning is perfectly fine, I'm just saying that you'll end up calling the method twice either way, one for the config and again for the layer. It's the internal logic that has to deal with it properly. Or am I missing something here?

I see your point. Yes, having two separate functions instead of one makes sense.